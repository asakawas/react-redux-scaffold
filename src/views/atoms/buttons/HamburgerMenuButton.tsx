import React, { FunctionComponent } from 'react';
import IconButton, { IconButtonProps } from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';

/**
 * The type of props of HamburgerMenuButton.
 */
export type HamburgerMenuButtonProps = Readonly<{
  /** The event handler called when the menu button is clicked */
  onHamburgerMenuButtonClick: IconButtonProps['onClick'];
}>;

const HamburgerMenuButton: FunctionComponent<HamburgerMenuButtonProps> = ({
  onHamburgerMenuButtonClick,
}) => (
  <IconButton edge="start" onClick={onHamburgerMenuButtonClick} aria-label="hamburger menu">
    <MenuIcon />
  </IconButton>
);

export default HamburgerMenuButton;
