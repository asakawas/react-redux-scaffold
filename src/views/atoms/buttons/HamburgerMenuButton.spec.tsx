import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import HamburgerMenuButton from './HamburgerMenuButton';

describe('HamburgerMenuButton', () => {
  test('is rendered as expected', () => {
    const { asFragment } = render(<HamburgerMenuButton onHamburgerMenuButtonClick={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("calls the passed handler when it's clicked", () => {
    const handler = jest.fn();
    const renderResult = render(<HamburgerMenuButton onHamburgerMenuButtonClick={handler} />);

    const tree = renderResult.getByLabelText('hamburger menu');
    fireEvent.click(tree);
    expect(handler).toHaveBeenCalledTimes(1);

    fireEvent.click(tree);
    expect(handler).toHaveBeenCalledTimes(2);
  });
});
