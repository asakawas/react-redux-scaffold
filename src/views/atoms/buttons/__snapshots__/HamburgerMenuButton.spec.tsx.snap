// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HamburgerMenuButton is rendered as expected 1`] = `
<DocumentFragment>
  <button
    aria-label="hamburger menu"
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeStart"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiIconButton-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
